stream.go

```diff
@@ -66,45 +72,182 @@ func (f *File) NewStreamWriter(sheet string) (*StreamWriter, error) {
 		File:    f,
 		Sheet:   sheet,
 		SheetID: sheetID,
+		replaceMap: map[string][]byte{
+			"XMLName": []byte{},
+		},
 	}
+	rsw.encoder = xml.NewEncoder(&rsw.SheetData)
 	rsw.SheetData.WriteString("<sheetData>")
 	return rsw, nil
 }
 
-// SetRow writes an array to streaming row by given worksheet name, starting
-// coordinate and a pointer to array type 'slice'. Note that, cell settings
-// with styles are not supported currently and after set rows, you must call the
-// 'Flush' method to end the streaming writing process. The following
-// shows the supported data types:
+// SetTable creates an Excel table for the StreamWriter. This function must be
+// called before the first SetRow.
 //
-//    int
-//    string
-//
-func (sw *StreamWriter) SetRow(axis string, slice interface{}) error {
+// The first SetRow will create the column names. These Column names must be
+// strings or an error will be returned.
+func (sw *StreamWriter) SetTable(format string) error {
```

Comment:
I propose to change the function name to `AddTable`. Also, It seems that there is no table range in the function parameters. Table range should be similar to [`AddTable`](https://github.com/360EntSecGroup-Skylar/excelize/blob/7358dca436f6ca5948a3f2865b14e828863d86a9/table.go#L53)

Reply:
`SetTable` must be done before `SetRow`. I used this approach because it is difficult to read the table headers once they are written to a temp file.

I agree it is a poor approach and the code is complicated.

Therefore I changed my approach. I now have an `AddTable` function as you suggested. See 34fca084014aa0ed4300791a177a70dc583766cc.

This was a major re-work. But I made a mechanism to read the temp file so we can get the table headers.

I now do a much better job of creating the final XML during `Flush`. We do exactly one allocation to fit the entire file contents, and we don't do any more processing. I found this dropped my total memory usage significantly.